snippets/country-localization.liquid

```diff
@@ -5,41 +5,129 @@
+      {{- localization.country.name }} |
+      {{ localization.country.currency.iso_code }}
+      {{ localization.country.currency.symbol -}}
     </span>
     {% render 'icon-caret' %}
   </button>
-  <div class="disclosure__list-wrapper" hidden>
-    <ul id="{{ localPosition }}List" role="list" class="disclosure__list list-unstyled">
-      {%- for country in localization.available_countries -%}
-        <li class="disclosure__item" tabindex="-1">
-          <a
-            class="link link--text disclosure__link caption-large focus-inset{% if country.iso_code == localization.country.iso_code %} disclosure__link--active{% endif %}"
-            href="#"
-            {% if country.iso_code == localization.country.iso_code %}
-              aria-current="true"
-            {% endif %}
-            data-value="{{ country.iso_code }}"
+  <div class="disclosure__list-wrapper country-selector" hidden>
+    <div class="country-selector__filter">
+      {% if localization.available_countries.size > 9 %}
+        <div class="field">
+          <input
+            class="country-filter__input field__input"
+            id="country-filter-input"
+            type="search"
+            name="country_filter"
+            value=""
+            placeholder="{{ 'localization.search' | t }}"
+            role="combobox"
+            aria-owns="country-results"
+            aria-controls="country-results"
+            aria-haspopup="listbox"
+            aria-autocomplete="list"
+            autocorrect="off"
+            autocomplete="off"
+            autocapitalize="off"
+            spellcheck="false"
+          >
+          <label class="field__label" for="country-filter-input">{{ 'localization.search' | t }}</label>
+          <button
+            type="reset"
+            class="country-filter__reset-button field__button hidden"
+            aria-label="{{ 'general.search.reset' | t }}"
           >
-            <span class="localization-form__currency">
-              {{- country.currency.iso_code }}
-              {{ country.currency.symbol }} |</span
+            <svg class="icon icon-close" aria-hidden="true" focusable="false">
+              <use xlink:href="#icon-reset">
+            </svg>
+          </button>
+          <div class="search__icon field__button motion-reduce" aria-label="{{ 'general.search.search' | t }}">
+            <svg class="icon icon-search" aria-hidden="true" focusable="false">
+              <use xlink:href="#icon-search">
+            </svg>
+          </div>
+        </div>
+      {% endif %}
+      <button
+        class="country-selector__close-button button--small link"
+        type="button"
+        aria-label="{{ 'accessibility.close' | t }}"
+      >
+        {% render 'icon-close' %}
+      </button>
+    </div>
+    <div
+      class="disclosure__list country-selector__list"
+      id="{{ localPosition }}-country-results"
+      aria-live="polite"
```

Comment:
Something that I notice as well is that when I open the country selector, it reads all the current list elements in there. 
So I open it, then it starts telling me all the currencies 😅 
The `aria-live="polite"` seem to be the issue. I think we could have it worked differently. Right now if I type as well something in the search, it's reading out all the list to me... I wonder if that is useful really to a user. Maybe if it's just a hidden div we update with a text saying `list updated` that isn't as invasive as mentioning all the items in the list. 

Not too sure, what's your take on it ?

Reply:
@ludoboludo which screen reader are you using? Using Mac's voice over I can't reproduce it reading all the countries and currencies in the list when typing.

Reply:
I agree, though, that it's not a friendly experience to have all the countries and currencies read out and will change it.

Reply:
Yeah it was with Voice Over on Safari. Were you doing voice over and chrome ? 🤔 Ill try again. 
Here is [a recording](https://screenshot.click/15-13-l3qaq-hpewi.mp4)